backend/mlarchive/archive/views.py

```diff
@@ -824,4 +824,54 @@ def get_context_data(self, **kwargs):
             date = date.replace(day=1)
         kwargs['subscribers'] = Subscriber.objects.filter(date=date)
         kwargs['date'] = date
-        return kwargs
\ No newline at end of file
+        return kwargs
+
+
+class ReportsMessagesView(View):
+    """Message Counts Report"""
+    
+    def get_message_stats(self, sdate, edate):
+        """Returns a tuple ( total messages, message counts as
+        list of tuples (listname, count))"""
+        messages = Message.objects.filter(
+            date__gte=sdate,
+            date__lt=edate,
+            email_list__private=False)
+        counter = Counter(messages.values_list('email_list__name', flat=True))
+        return (messages.count(), sorted(counter.items()))
```

Comment:
This will work, but it unless email_list__name can be reached only by indexes, it's going to be a bit of a memory hog, particularly if someone provides dates that are very far apart in time (You'd potentially pull the entire archive into memory).

Consider, instead, annotating with using django.db.models Count - (though that also would need good index management to not be abusive, I think?

In any case - please run some tests and make sure that this turns out to be reasonable efficient.

Reply:
Even when running over the entire archive, this process takes 1-2 seconds and uses 260MB RAM, which seems reasonable.